bootstrap/terraform/main.tf

```diff
@@ -87,30 +88,41 @@ module "ebs_csi_driver_irsa" {
 
 module "eks" {
   source  = "terraform-aws-modules/eks/aws"
-  version = "~> 19.13"
+  version = "~> 20.0"
 
   cluster_name                   = local.name
   cluster_version                = local.cluster_version
   cluster_endpoint_public_access = true
   kms_key_enable_default_policy  = true
 
+  # Give the Terraform identity admin access to the cluster
+  # which will allow resources to be deployed into the cluster
+  enable_cluster_creator_admin_permissions = true
+
+  vpc_id     = module.vpc.vpc_id
+  subnet_ids = module.vpc.private_subnets
+
   cluster_addons = {
     aws-ebs-csi-driver = {
+      most_recent              = true
       service_account_role_arn = module.ebs_csi_driver_irsa.iam_role_arn
     }
-    coredns    = {}
-    kube-proxy = {}
-    vpc-cni    = {}
+    coredns = {
+      most_recent = true
+    }
+    kube-proxy = {
+      most_recent = true
+    }
+    vpc-cni = {
+      most_recent = true
```

Comment:
Curious, what does `most_recent = true` does in opposed of omitting it?

Reply:
the module docs uses it, I followed the docs during the upgrade:

https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest#usage

Reply:
Now I really want to know, I will ask around.

Reply:
Here is the answer, with `coredns = {}` the module will apply the default version returned by the describe-addon api. See https://github.com/terraform-aws-modules/terraform-aws-eks/blob/59024ebd0aeacad2c1a47d7c8dad123042814dec/main.tf#L490.
By specifying `most_recent=true` you are overriding and asking for the latest version available which may or may not be the default version returned.